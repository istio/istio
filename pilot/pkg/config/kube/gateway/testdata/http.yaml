apiVersion: gateway.networking.k8s.io/v1beta1
kind: GatewayClass
metadata:
  name: istio
spec:
  controllerName: istio.io/gateway-controller
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: Gateway
metadata:
  name: gateway
  namespace: istio-system
spec:
  addresses:
  - value: istio-ingressgateway
    type: Hostname
  gatewayClassName: istio
  listeners:
  - name: default
    hostname: "*.domain.example"
    port: 80
    protocol: HTTP
    allowedRoutes:
      namespaces:
        from: All
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: HTTPRoute
metadata:
  name: http
  namespace: default
spec:
  parentRefs:
  - name: gateway
    namespace: istio-system
  hostnames: ["first.domain.example", "another.domain.example"]
  rules:
  - matches:
    - path:
        type: PathPrefix
        value: /get
      headers:
      - name: my-header
        value: some-value
        type: Exact
    filters:
    - type: RequestHeaderModifier
      requestHeaderModifier:
        add:
        - name: my-added-header
          value: added-value
        remove: [my-removed-header]
    - type: ResponseHeaderModifier
      responseHeaderModifier:
        add:
        - name: my-added-resp-header
          value: added-resp-value
        remove: [my-removed-header]
    backendRefs:
    - name: httpbin
      port: 80
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: HTTPRoute
metadata:
  name: http2
  namespace: default
spec:
  parentRefs:
  - name: gateway
    namespace: istio-system
  hostnames: ["second.domain.example"]
  rules:
  - matches:
    - path:
        type: PathPrefix
        value: /second
    backendRefs:
    - name: httpbin-second
      port: 80
  - matches:
    - path:
        type: PathPrefix
        value: /
    backendRefs:
    - name: httpbin-wildcard
      port: 80
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: HTTPRoute
metadata:
  name: redirect
  namespace: default
spec:
  parentRefs:
  - name: gateway
    namespace: istio-system
  rules:
  - filters:
    - type: RequestRedirect
      requestRedirect:
        port: 8080
        statusCode: 302
        scheme: https
        path:
          type: ReplaceFullPath
          replaceFullPath: /replace-full
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: HTTPRoute
metadata:
  name: redirect-prefix-replace
  namespace: default
spec:
  parentRefs:
  - name: gateway
    namespace: istio-system
  hostnames: ["redirect.domain.example"]
  rules:
  - matches:
    - path:
        type: PathPrefix
        value: /original
    filters:
    - type: RequestRedirect
      requestRedirect:
        port: 8080
        statusCode: 302
        scheme: https
        path:
          type: ReplacePrefixMatch
          replacePrefixMatch: /replacement
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: HTTPRoute
metadata:
  name: rewrite
  namespace: default
spec:
  parentRefs:
  - name: gateway
    namespace: istio-system
  rules:
  - name: route1
    matches:
    - path:
        type: PathPrefix
        value: /prefix-original
    filters:
    - type: URLRewrite
      urlRewrite:
        hostname: "new.example.com"
        path:
          type: ReplacePrefixMatch
          replacePrefixMatch: "/replacement"
    backendRefs:
    - name: httpbin
      port: 80
  - matches:
    - path:
        type: PathPrefix
        value: /prefix-to-be-removed
    filters:
    - type: URLRewrite
      urlRewrite:
        path:
          type: ReplacePrefixMatch
          replacePrefixMatch: ""
    backendRefs:
    - name: httpbin
      port: 80
  - matches:
    - path:
        type: PathPrefix
        value: /full-original
    filters:
    - type: URLRewrite
      urlRewrite:
        hostname: "new.example.com"
        path:
          type: ReplaceFullPath
          replaceFullPath: "/replacement"
    backendRefs:
    - name: httpbin
      port: 80
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: HTTPRoute
metadata:
  name: mirror
  namespace: default
spec:
  parentRefs:
  - name: gateway
    namespace: istio-system
  rules:
  - filters:
    - type: RequestMirror
      requestMirror:
        fraction:
          numerator: 4
          denominator: 8
        backendRef:
          name: httpbin-mirror
          port: 80
    - type: RequestMirror
      requestMirror:
        percent: 80
        backendRef:
          name: httpbin-second
          port: 80
    backendRefs:
    - name: httpbin
      port: 80
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: HTTPRoute
metadata:
  name: http-not-selected
  namespace: default
spec:
  parentRefs:
  - name: gateway
    namespace: istio-system
  hostnames: ["should.not.select"]
  rules:
  - matches:
    - path:
        type: PathPrefix
        value: /get
    backendRefs:
    - name: httpbin-bad
      port: 80
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: HTTPRoute
metadata:
  name: http-timeout-request
  namespace: default
spec:
  parentRefs:
  - name: gateway
    namespace: istio-system
  hostnames: ["timeout.domain.example"]
  rules:
  - matches:
    - path:
        type: PathPrefix
        value: /get
    backendRefs:
    - name: httpbin
      port: 80
    timeouts:
      request: 1ms
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: HTTPRoute
metadata:
  name: http-timeout-backend-request
  namespace: default
spec:
  parentRefs:
  - name: gateway
    namespace: istio-system
  hostnames: ["timeout-backend.domain.example"]
  rules:
  - matches:
    - path:
        type: PathPrefix
        value: /get
    backendRefs:
    - name: httpbin
      port: 80
    timeouts:
      request: 2ms
      backendRequest: 1ms
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: HTTPRoute
metadata:
  name: http-retry-request
  namespace: default
spec:
  parentRefs:
    - name: gateway
      namespace: istio-system
  hostnames: ["retry.domain.example"]
  rules:
    - matches:
        - path:
            type: PathPrefix
            value: /explicit
      backendRefs:
        - name: httpbin
          port: 80
      retry:
        attempts: 3
        codes:
        - 503
        - 429
    - matches:
        - path:
            type: PathPrefix
            value: /empty
      backendRefs:
        - name: httpbin
          port: 80
      retry: {}
    - matches:
        - path:
            type: PathPrefix
            value: /disable
      backendRefs:
        - name: httpbin
          port: 80
      retry:
        attempts: 0
