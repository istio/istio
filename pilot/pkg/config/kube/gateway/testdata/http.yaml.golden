apiVersion: networking.istio.io/v1
kind: Gateway
metadata:
  annotations:
    internal.istio.io/gateway-semantics: gateway
    internal.istio.io/gateway-service: istio-ingressgateway.istio-system.svc.domain.suffix
    internal.istio.io/parents: Gateway/gateway/default.istio-system
  creationTimestamp: null
  name: gateway-istio-autogenerated-k8s-gateway-default
  namespace: istio-system
spec:
  servers:
  - hosts:
    - '*/*.domain.example'
    port:
      name: default
      number: 80
      protocol: HTTP
---
apiVersion: networking.istio.io/v1
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parents: HTTPRoute/http.default
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: http-0-istio-autogenerated-k8s-gateway
  namespace: default
spec:
  gateways:
  - istio-system/gateway-istio-autogenerated-k8s-gateway-default
  hosts:
  - first.domain.example
  http:
  - headers:
      request:
        add:
          my-added-header: added-value
        remove:
        - my-removed-header
      response:
        add:
          my-added-resp-header: added-resp-value
        remove:
        - my-removed-header
    match:
    - headers:
        my-header:
          exact: some-value
      uri:
        prefix: /get
    name: default.http.0
    route:
    - destination:
        host: httpbin.default.svc.domain.suffix
        port:
          number: 80
---
apiVersion: networking.istio.io/v1
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parents: HTTPRoute/http.default
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: http-1-istio-autogenerated-k8s-gateway
  namespace: default
spec:
  gateways:
  - istio-system/gateway-istio-autogenerated-k8s-gateway-default
  hosts:
  - another.domain.example
  http:
  - headers:
      request:
        add:
          my-added-header: added-value
        remove:
        - my-removed-header
      response:
        add:
          my-added-resp-header: added-resp-value
        remove:
        - my-removed-header
    match:
    - headers:
        my-header:
          exact: some-value
      uri:
        prefix: /get
    name: default.http.0
    route:
    - destination:
        host: httpbin.default.svc.domain.suffix
        port:
          number: 80
---
apiVersion: networking.istio.io/v1
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parents: HTTPRoute/http-timeout-backend-request.default
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: http-timeout-backend-request-0-istio-autogenerated-k8s-gateway
  namespace: default
spec:
  gateways:
  - istio-system/gateway-istio-autogenerated-k8s-gateway-default
  hosts:
  - timeout-backend.domain.example
  http:
  - match:
    - uri:
        prefix: /get
    name: default.http-timeout-backend-request.0
    route:
    - destination:
        host: httpbin.default.svc.domain.suffix
        port:
          number: 80
    timeout: 0.001s
---
apiVersion: networking.istio.io/v1
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parents: HTTPRoute/http-timeout-request.default
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: http-timeout-request-0-istio-autogenerated-k8s-gateway
  namespace: default
spec:
  gateways:
  - istio-system/gateway-istio-autogenerated-k8s-gateway-default
  hosts:
  - timeout.domain.example
  http:
  - match:
    - uri:
        prefix: /get
    name: default.http-timeout-request.0
    route:
    - destination:
        host: httpbin.default.svc.domain.suffix
        port:
          number: 80
    timeout: 0.001s
---
apiVersion: networking.istio.io/v1
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parents: HTTPRoute/http2.default
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: http2-0-istio-autogenerated-k8s-gateway
  namespace: default
spec:
  gateways:
  - istio-system/gateway-istio-autogenerated-k8s-gateway-default
  hosts:
  - second.domain.example
  http:
  - match:
    - uri:
        prefix: /second
    name: default.http2.0
    route:
    - destination:
        host: httpbin-second.default.svc.domain.suffix
        port:
          number: 80
  - match:
    - uri:
        prefix: /
    name: default.http2.1
    route:
    - destination:
        host: httpbin-wildcard.default.svc.domain.suffix
        port:
          number: 80
---
apiVersion: networking.istio.io/v1
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parents: HTTPRoute/mirror.default,HTTPRoute/redirect.default,HTTPRoute/rewrite.default
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: mirror-0-istio-autogenerated-k8s-gateway
  namespace: default
spec:
  gateways:
  - istio-system/gateway-istio-autogenerated-k8s-gateway-default
  hosts:
  - '*'
  http:
  - match:
    - uri:
        prefix: /prefix-to-be-removed
    name: default.rewrite.1
    rewrite:
      uri: /
    route:
    - destination:
        host: httpbin.default.svc.domain.suffix
        port:
          number: 80
  - match:
    - uri:
        prefix: /prefix-original
    name: route1
    rewrite:
      authority: new.example.com
      uri: /replacement
    route:
    - destination:
        host: httpbin.default.svc.domain.suffix
        port:
          number: 80
  - match:
    - uri:
        prefix: /full-original
    name: default.rewrite.2
    rewrite:
      authority: new.example.com
      uriRegexRewrite:
        match: /.*
        rewrite: /replacement
    route:
    - destination:
        host: httpbin.default.svc.domain.suffix
        port:
          number: 80
  - mirrors:
    - destination:
        host: httpbin-mirror.default.svc.domain.suffix
        port:
          number: 80
      percentage:
        value: 50
    - destination:
        host: httpbin-second.default.svc.domain.suffix
        port:
          number: 80
      percentage:
        value: 80
    name: default.mirror.0
    route:
    - destination:
        host: httpbin.default.svc.domain.suffix
        port:
          number: 80
  - name: default.redirect.0
    redirect:
      port: 8080
      redirectCode: 302
      scheme: https
      uri: /replace-full
---
apiVersion: networking.istio.io/v1
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parents: HTTPRoute/redirect-prefix-replace.default
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: redirect-prefix-replace-0-istio-autogenerated-k8s-gateway
  namespace: default
spec:
  gateways:
  - istio-system/gateway-istio-autogenerated-k8s-gateway-default
  hosts:
  - redirect.domain.example
  http:
  - match:
    - uri:
        prefix: /original
    name: default.redirect-prefix-replace.0
    redirect:
      port: 8080
      redirectCode: 302
      scheme: https
      uri: '%PREFIX()%/replacement'
---
